app/controllers/hotspots_controller.rb

```diff
@@ -11,29 +15,28 @@ def new
     
     def create
         @hotspot = Hotspot.create(hotspot_params)
-
         if @hotspot.valid?
             flash[:notice] = "Hotspot created succesfully."
         else
             flash[:notice] = "Hotspot could not be saved."
         end
-        
+        @hotspot.save
         redirect_to hotspots_path
     end
     
     def edit
         @hotspot = Hotspot.find(params[:id])
+        @hotspot.save
     end
     
     def update
         @hotspot = Hotspot.find(params[:id])
         if @hotspot.update_attributes(hotspot_params)
-     
             flash[:notice] = "Hotspot updated succesfully."
         else
             flash[:notice] = "There was an error updating the hotspot."
         end
-        
+        @hotspot.save
```

Comment:
No need for this, it's being save on line 34 when you call `@hotspot.update_attributes`